docs/contribute/api-docs-contribution/guidelines.md

```diff
@@ -0,0 +1,575 @@
+  query_and_fetch = "query_and_fetch"
+}
+```
+
+The compiler automatically extracts these descriptions and formats them as lists in the property documentation:
+
+- `query_then_fetch`: Searches across all fields using default settings
+- `dfs_query_then_fetch`: Performs distributed frequency scoring before fetching results
+- `query_and_fetch`: Fetches results immediately without scoring optimization
+:::
+
+::::
+
+
+## Add examples
+
+Examples help users understand how to use your API with realistic request and response data. Well-written descriptions transform examples from code snippets into effective learning tools.
+
+Here are some principles for effective examples:
+
+- **Use realistic data:** Provide examples that reflect actual use cases rather than placeholder values
+- **Write clear summaries:** Introduce the example with a meaningful summary that explains its purpose
+- **Write clear descriptions:** Explain what the example accomplishes in more detail and why it's useful
+- **Include edge cases:** Show how to handle optional parameters, error conditions, and different response types
+- **Point out key details:** Highlight important aspects users might miss
+- **Show variations:** Demonstrate alternative approaches or related concepts
+- **Use descriptive names:** Choose meaningful file names and summaries that indicate the example's purpose
+- **Provide realistic response bodies:** Each response body should have a realistic example. It must not contain any sensitive or confidential data
+- **Include success responses:** Include at least one example for each success response (HTTP 200)
+- **Schema flexibility:** Examples don't need to strictly validate against schemas, but should represent valid responses
+
+#### Generated examples
+
+If you don't provide examples, Bump.sh automatically generates them from your API schema. Because it's hard to randomly generate meaningful examples, this has been disabled for Elasticsearch APIs.
+
+It's preferable to include curated examples in your OpenAPI document with a realistic combination of property values. You can provide more than one example object in the examples field, so consider adding examples that reflect the most common use cases.
+
+Bump.sh also generates a single curl example for each API. To override that generated example and optionally provide an equivalent Console or language-specific example, use the `x-codeSamples` extension described in [Custom code examples in Bump.sh](https://docs.bump.sh/help/specification-support/doc-code-samples).
+
+:::{tip}
+When you use validation tools to check your API specification, examples are helpful for ensuring that you haven't missed things like nullable data types.
+
+Learn more:
+
+- [The example object](https://spec.openapis.org/oas/latest#example-object)
+- [Add examples](https://swagger.io/docs/specification/adding-examples)
+:::
+
+### Examples
+
+**File-based examples** use a structured folder approach alongside your API specification:
```

Comment:
> **File-based examples** use a structured folder approach alongside your API specification:

This is an Elasticsearch-specific implementation detail, so probably good to put it into an Elasticsearch-specific tab like the other examples prior to this.